stytch/src/main/kotlin/com/stytch/java/consumer/api/m2m/M2M.kt

```diff
@@ -243,17 +238,50 @@ internal class M2MImpl(
         data: AuthenticateTokenRequest,
         callback: (StytchResult<AuthenticateTokenResponse>) -> Unit,
     ) {
-        coroutineScope.launch {
-            callback(authenticateToken(data))
-        }
+        coroutineScope.launch { callback(authenticateToken(data)) }
     }
 
     override suspend fun authenticateTokenCompletable(
         data: AuthenticateTokenRequest,
     ): CompletableFuture<StytchResult<AuthenticateTokenResponse>> {
-        return coroutineScope.async {
-            authenticateToken(data)
-        }.asCompletableFuture()
+        return coroutineScope.async { authenticateToken(data) }.asCompletableFuture()
+    }
+
+    companion object {
+        @JvmStatic
+        fun performAuthorizationCheck(
+            hasScopes: List<String>,
+            requiredScopes: List<String>,
+        ) {
+            val clientScopes: MutableMap<String, MutableSet<String>> = mutableMapOf()
+            for (scope in hasScopes) {
```

Comment:
You could make these for loops a little more idiomatic like the following, but I'm definitely not pressed about it:

```
hasScopes.forEach { scope ->
    val (action, resource) = if (scope.contains(":")) {
        scope.split(":")
    } else {
        listOf(scope, "-")
    }
    clientScopes.computeIfAbsent(action) { mutableSetOf() }.add(resource)
}
```